pkg/util/tar_utils.go

```diff
@@ -65,7 +67,7 @@ func unpackTar(tr *tar.Reader, path string) error {
 				if err := os.MkdirAll(target, mode); err != nil {
 					return err
 				}
-			} else {
+				// In some cases, MkdirAll doesn't change the permissions, so run Chmod
```

Comment:
It's usually umask related, but I suppose if the dirs already exist it wouldn't change the permissions either